src/Microsoft.Health.Fhir.Core/Features/Persistence/ResourceWrapper.cs

```diff
@@ -119,5 +119,16 @@ public void UpdateSearchIndices(IReadOnlyCollection<SearchIndexEntry> searchIndi
         {
             SearchIndices = searchIndices;
         }
+
+        public virtual bool SearchIndicesEqual(IReadOnlyCollection<SearchIndexEntry> otherIndices)
```

Comment:
If one of the lists has two search index entries for "name" and the other has only one entry for "name", we will still end up returning true here. Is that the expected behavior?